elytron/src/main/java/org/wildfly/extension/elytron/AuthenticationClientDefinitions.java

```diff
@@ -317,7 +318,12 @@ static ResourceDefinition getAuthenticationClientDefinition() {
                         try {
                             CredentialSource cs = sourceSupplier.get();
                             if (cs != null) {
-                                return c.usePassword(cs.getCredential(PasswordCredential.class).getPassword());
+                                PasswordCredential passCredential = cs.getCredential(PasswordCredential.class);
+                                String alias = credentialReference.hasDefined(CredentialReference.ALIAS) ? credentialReference.get(CredentialReference.ALIAS).asString() : null;
+                                if (passCredential == null && alias != null && alias.length() > 0) {
+                                    throw ROOT_LOGGER.credentialDoesNotExist(alias, PasswordCredential.class.getName());
+                                }
+                                return c.usePassword(passCredential.getPassword());
```

Comment:
The [alias is required](https://github.com/wildfly/wildfly-core/blob/6f65f331de79123468cf69d5a67a68d92a591feb/controller/src/main/java/org/jboss/as/controller/security/CredentialReference.java#L107) in this situation, but for thoroughly checking, I agree it will be better to check the case as well. :)